### PR TITLE
RPE-691: Ignore app versions not compliant with semver specs

### PR DESCRIPTION
**Pull request by laszlorozsas**
Ticket: https://deliveroo.atlassian.net/browse/RPE-691

Related spec updates: https://github.com/deliveroo/determinator-standard-tests/pull/15

The Determinator gem raises an exception on `app_version`s with the following format: 2021.11.05. It should just return false on that determination if the input is invalid.

According to the semantic versioning specs none of the components of the version number should have any trailing zeros. So the above version should be 2021.11.5.

Setting the above non-compliant app version caused the following issue:
Sentry: https://sentry.io/organizations/deliveroo/issues/2813035755/?project=1460916&query=is%3Aunresolved

### CODE REVIEW

File: lib/determinator/control.rb [hunk 1]
```diff
@@ -245,7 +245,8 @@ def has_any_app_version?(required, present)
         !v.match?(Semantic::Version::SemVerRegexp)
       end
       invalid_groups = required.flatten.select do |v|
-        !v.match?(/\d/)
+        without_operator = v.gsub(/>=|>|<=|<|~>/, '')
```
Comment (me) [inline lib/determinator/control.rb:248]: Might make sense to do a split instead [like Semantic::Version does](https://github.com/jlindsey/semantic/blob/master/lib/semantic/version.rb#L96)?

**Review by me — COMMENTED** [review]
I think this makes sense! Do we have tickets to do this for the Go and Scala libraries as well? They'll need that when the standard tests are updated.

**Comment by laszlorozsas** [comment]
> I think this makes sense! Do we have tickets to do this for the Go and Scala libraries as well? They'll need that when the standard tests are updated.

I checked the golang library and the new test passed without any update in the library
I don't have experience with Scala but I asked for some help so hopefully I'll soon be able to verify that as well

**Comment by laszlorozsas** [comment]
> I think this makes sense! Do we have tickets to do this for the Go and Scala libraries as well? They'll need that when the standard tests are updated.

I also checked the Scala library and the new test passed without any update in the library

**Review by me — APPROVED** [review]
LGTM!

**Pull request merged**